src/window.cpp

```diff
@@ -1944,10 +1944,15 @@ void ResetWindowSystem()
 
 static void DecreaseWindowCounters()
 {
+	static byte hundredth_tick_timeout = 100;
+
 	if (_scroller_click_timeout != 0) _scroller_click_timeout--;
+	if (hundredth_tick_timeout != 0) hundredth_tick_timeout--;
 
 	Window *w;
 	FOR_ALL_WINDOWS_FROM_FRONT(w) {
+		if (!_network_dedicated && hundredth_tick_timeout == 0) w->OnHundredthTick();
```

Comment:
OnHundredthTick() is now called before ProcessScheduledInvalidations().

That looks dangerous to me.